coap-core/src/main/java/com/mbed/coap/transport/udp/IDatagramSocket.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright (C) 2011-2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.mbed.coap.transport.udp;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+
+public interface IDatagramSocket {
```

Comment:
I'd recommend to rename into `BlockingSocket`

Reply:
This interface could be quite simple, for example:
```
public interface BlockingSocket extends Closeable {

    void send(DatagramPacket datagramPacket) throws IOException;

    void receive(DatagramPacket datagramPacket) throws IOException;

    InetSocketAddress getBoundAddress();

}
```

Reply:
done.